.gitbook/5-oracles/1-operate-oracle-nodes/4-oracle-registration.md

```diff
@@ -0,0 +1,96 @@
+| oracle-commission-rate            | required    | The desired initial oracle commission rate                                                           |
+| oracle-commission-max-rate        | required    | The maximum oracle commission rate. The oracle commission rate cannot be greater than this max rate. |
+| oracle-commission-max-change-rate | required    | The maximum rate that an oracle can change once. It will be reset 24 hours after the last change.    |
+| oracle-endpoint                   | optional    | The endpoint of oracle to be used                                                                    |
+
+With the above arguments, you can now request to register your oracle to Panacea.
+
+```bash
+docker run \
+    --device /dev/sgx_enclave \
+    --device /dev/sgx_provision \
+    -v ${ANY_DIR_ON_HOST}:/oracle \
+    ghcr.io/medibloc/panacea-oracle:latest \
+    ego run /usr/bin/oracled register-oracle \ 
+    --trusted-block-height ${HEIGHT} \
+    --trusted-block-hash ${HASH} \
+    --oracle-commission-rate 0.1 \
+    --oracle-commission-max-rate 0.3 \
+    --oracle-commission-max-change-rate 0.01 \
+    --oracle-endpoint "<your-oracle-endpoint>"
+```
+
+Then, a new key pair, called `Node Key`, and its remote report will be generated.
+The `Node Key` is used to be shared the oracle private key.
+Since the oracle private key must be shared in a very secure way, the key will be encrypted with the public key of the `Node Key` so that only the oracle can decrypt it.
+
+{% hint style="danger" %}
+The private key of `Node Key` is also stored in secure way as a sealed file, named `node_priv_key.sealed` in default.
+This `node_priv_key.sealed` file is the only clue to decrypt the encrypted oracle private key, we recommend to manage it very carefully in case that you need to restore the oracle private key again later.
+{% endhint %}
+
+If you have tried to request `register-oracle` before, previously generated `node_priv_key.sealed` would exist.
+And the app will ask, 
+
+```
+There is an existing node key.
+Are you sure to delete and re-generate node key?
+```
+
+If you're sure to re-generate the `Node Key` and re-request oracle registration, please enter `y`.
+Or, we recommend to back up the existing `node_priv_key.sealed` file.
+
+## Subscribe Approval of Registration
+
+If the transaction for request of oracle registration succeeds, oracle will start to subscribe `ApproveOracleRegistrationEvent`.
+Other oracles already registered will do some verifications of this registration by checking if:
+- correct version of oracle binary is used
+- the oracle is running inside an enclave 
+- the `Node Key` is generated inside the enclave
+- the trusted block information is valid
+
+When the registration is verified successfully, other oracles will send a transaction for approval of the oracle registration.
+The oracle catch the approval transaction, and then the oracle start to retrieve the oracle private key.
```

Comment:
How about "subscribe approval event" rather than "catch transaction"? I'm not sure which one is better, but I just give you an opinion.

Reply:
I thought "subscribe" means listening and waiting for events, so I used "catch" for detecting the emitted event.
Or, detect the approval event? I'll think about this expression. ty